torchvision/prototype/transforms/_auto_augment.py

```diff
@@ -361,7 +352,7 @@ def __init__(
     def forward(self, *inputs: Any) -> Any:
         sample = inputs if len(inputs) > 1 else inputs[0]
 
-        id, image = self._extract_image(sample)
+        sample_flat, sample_spec, idx_flat, image = self._flatten(sample)
```

Comment:
I appologize if I'm missing the context, If I understand correctly the code, you fetch the first image from sample with `_flatten` and below in the code the image is transformed and written back to sample.
Why do we transform only the first image ?

Reply:
More generally question, why can't we restructure the implementation such that we use `forward` to 
inspect sample and ensure that only images (and/or only one image in sample) go further into `_transform`. In `_transform` we do the necessary AA ops. What do you think ?